packages/query/src/actions.ts

```diff
@@ -188,6 +188,31 @@ export async function getTypeData(
   return { params, fields };
 }
 
+function mergeTypeRow<T extends Record<number, MappableType>>(
```

Comment:
Looks like this function is meant to be used in a reduce context.
What do you think about moving the reduce call into this function and renaming it to `mergeTypeRows`?
Adding some basic tests for the resulting function would also be nice to have, especially given that `typeRows` are typed as `any` atm.